From 6a21ce2eb1d7530f20803c6077ad0dfe16658e09 Mon Sep 17 00:00:00 2001
From: Hiromitsu Yamasaki <hiromitsu.yamasaki.ym@renesas.com>
Date: Tue, 28 Jun 2016 11:41:47 +0900
Subject: [PATCH] spi: sh-msiof: Fix calculation of the division value

This patch is corresponding to the prohibitions of the hardware manual
for R-Car Gen2/Gen3 SoCs.
In case of Gen2/Gen3, BRDV[2:0]=B'111 is valid only
when the BRPS[4:0] bits are set to B'00000 or B'00001.

Signed-off-by: Hiromitsu Yamasaki <hiromitsu.yamasaki.ym@renesas.com>
---
 drivers/spi/spi-sh-msiof.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/spi/spi-sh-msiof.c b/drivers/spi/spi-sh-msiof.c
index 1317216..6969c8b 100644
--- a/drivers/spi/spi-sh-msiof.c
+++ b/drivers/spi/spi-sh-msiof.c
@@ -344,6 +344,7 @@ static int msiof_rcar_is_gen2(struct device *dev)
 
 	return of_device_is_compatible(node, "renesas,msiof-r8a7790") ||
 		of_device_is_compatible(node, "renesas,msiof-r8a7791") ||
+		of_device_is_compatible(node, "renesas,msiof-r8a7792") ||
 		of_device_is_compatible(node, "renesas,msiof-r8a7793") ||
 		of_device_is_compatible(node, "renesas,msiof-r8a7794");
 }
-- 
1.9.1

